src/eq.js

```diff
@@ -0,0 +1,261 @@
+   *      mouseOnHandle(this.x,this.y) ? (this.y=mouseY, makeAdjustment(this.ind,this.y,this.x)) : true;
+   *    } else{
+   *      mouseOnHandle(this.x,this.y) ? (this.x=mouseX, this.y=mouseY, makeAdjustment(this.ind,this.y,this.x)) : true;
+   *    }
+
+   *    if (this.y < 1) {this.y = 1;}
+   *    else if (this.y >255) { this.y = 255;}
+   *    else if (this.x < 1) { this.x = 1;}
+   *    else if (this.x > 709) { this.x = 709;} 
+   *  }
+   *  this.toggle = function () {
+   *    eq.toggleBand(this.ind);
+   *    eq.bands[this.ind].toggle ? this.c = color(255) : this.c = color(30);
+   *  }
+   * }
+
+
+
+   *  function mouseClicked(){
+   *   //console.log(cpts.length);
+   * for (var i = 0; i < cpts.length; i++) {
+    
+   * if (mouseX > cpts[i].x - circleSize && mouseX < cpts[i].x + circleSize &&
+   *       mouseY > cpts[i].y - circleSize && mouseY < cpts[i].y + circleSize){
+   *    cpts[i].toggle();
+   *  } 
+   * }
+      
+   *  }
+
+   *  function makeAdjustment(band,y,x){
+   *  eq.setBand(band,"mod",map(y,256,0, -40, 40), map(x,0,710,0,22050));
+   * }
+
+   * function mouseOnHandle(x,y){
+   *    if (mouseX > x - circleSize && mouseX < x + circleSize &&
+   *     mouseY > y - circleSize && mouseY < y + circleSize &&
+   *     mouseIsPressed){
+   *      return true;
+   *    }
+   * else{
+   *    return false;
+   *  }
+  * }
+   *  </code></div>
+   */
+
+
+  p5.EQ = function(_eqsize) {
+
+    //default size is 8 band EQ
+    _eqsize = _eqsize || 8;
+
+    Effect.call(this);
+    
+    this.bands = [];
+
+    /**
+      *  The p5.EQ is built with
+      *  <a href="http://www.w3.org/TR/webaudio/#BiquadFilterNode">
+      *  Web Audio BiquadFilter Node</a>.
+      *  
+      *  @property biquadFilter
+      *  @type {Object}  Web Audio Delay Node
+      *
+      *  @{param} toggle {boolean} On/of switch for band, true == on
+    */
+    for (var i = 0; i<_eqsize; i++){
+      this.bands.push(this.ac.createBiquadFilter());
+      this.bands[i].type = 'peaking';
+      this.bands[i].frequency.value = i*3150;
+      this.bands[i].Q.value = 5;
+      this.toggle = true;
+      if (i>0){
+        this.bands[i-1].connect(this.bands[i]);
+      }
+    }
+    this.input.connect(this.bands[0]);
+    this.bands[_eqsize-1].connect(this.output);
+  };
+
+  p5.EQ.prototype = Object.create(Effect.prototype);
+
+  p5.EQ.prototype.process = function (src) {
+    src.connect(this.input);
+  };
+
+  /**
+   *  @{method} setBand Make adjustments to individual bands of the EQ
+   *
+   *  @{param} band {number} Band to be modified
+   *  @{param} option {string} Modify function (toggle, mod, type);
+   *  @{param} param1 {number} Set the frequency of a band w/ usage: "mod"
+   *  @{param} param2 {number} Set the Q value (resonance) of a band w/usage: "mod"
+   *  @{param} param1 {string} Set the type of the band filter w/ usage: "type"
+   */
+  p5.EQ.prototype.setBand = function (band, option, param1, param2) {
```

Comment:
I do agree that EQ should do as little re-implementation as possible-- The problem that I have run into with using [p5.Filter](https://p5js.org/reference/#/p5.Filter) to control the bands is that the are the [BiquadFilters](https://www.w3.org/TR/webaudio/#the-biquadfilternode-interface) are wrapped in [Gain](https://www.w3.org/TR/webaudio/#GainNode) nodes (due to [p5.Effect](https://github.com/processing/p5.js-sound/blob/master/src/effect.js#L10) and the input / output standard format of p5.js-sound.

All these issues extra Gain nodes seem to affect the accuracy of the EQ (boosting the overall volume, making it impossible to filter out highs...etc). I think this problem could be addressed either with different methods or by using the design you have indicated above, but with methods such as `.freq()` implemented in `p5.EQ`(because they have to work on the raw biquad filter). what do you think?